src/texteff.cpp

```diff
@@ -42,13 +42,13 @@ TextEffectID AddTextEffect(StringID msg, int center, int y, uint8_t duration, Te
 {
 	if (_game_mode == GM_MENU) return INVALID_TE_ID;
 
-	TextEffectID i;
-	for (i = 0; i < _text_effects.size(); i++) {
-		if (_text_effects[i].string_id == INVALID_STRING_ID) break;
+	auto it = std::find_if(std::begin(_text_effects), std::end(_text_effects), [](const TextEffect &te) { return te.string_id == INVALID_STRING_ID; });
+	if (it == std::end(_text_effects)) {
+		if (_text_effects.size() == INVALID_TE_ID) return INVALID_TE_ID;
```

Comment:
Please add a comment of the likes: `INVALID_TE_ID` index is reserved and can't be used.

Or anything to explain why you are comparing a size with an index :D